cmd/argocd/commands/login.go

```diff
@@ -259,11 +259,12 @@ func oauth2Login(ctx context.Context, port int, oidcSettings *settingspkg.OIDCCo
 	}
 	fmt.Printf("Performing %s flow login: %s\n", grantType, url)
 	time.Sleep(1 * time.Second)
-	err = open.Run(url)
+	err = open.Start(url)
```

Comment:
can you tell me what this changes does please?

Reply:
Thank you for confirming.

`open.Run()` is a blocking function. So it may wait for the close of the web browser which was launched by the function. In this case, the temporary HTTP server does not start and the callback request (http://localhost:8085/auth/callback) will not be accepted.

This behavior is environment-dependent.

In my environment(Ubuntu 18.04 + Chromium), it will be caused when no other web browser is launched. In this situation, `open.Run()` launches a new web browser and wait for the close of it. So the login flow does not finish.

But when other web browser already exists, `open.Run()` just creates a new tab and does not wait. So the login flow will complete.

`open.Start()` is a non-blocking function. So the temporary HTTP server always starts and the login flow succeeds.

ref: https://godoc.org/github.com/skratchdot/open-golang/open